src/routes/workspaces.py

```diff
@@ -303,3 +317,50 @@ async def get_my_sql_warehouse_prefix(
 
     logger.info(f"Retrieved SQL warehouse prefix for user {username}")
     return response
+
+
+@router.get(
+    "/me/namespace-prefix",
+    response_model=NamespacePrefixResponse,
+    summary="Get governance namespace prefix",
+    description=(
+        "Get the governance namespace prefix for the authenticated user. "
+        "Optionally provide a tenant (group) to get the tenant namespace prefix; "
+        "membership is required."
+    ),
+)
+async def get_namespace_prefix(
+    authenticated_user: Annotated[KBaseUser, Depends(auth)],
+    request: Request,
+    tenant: Annotated[str | None, Query(description="Optional tenant (group) name", min_length=1)] = None,
+):
+    """Return governance namespace prefix for user or specified tenant.
+
+    - Without tenant: returns user's prefix, e.g. `u_<user>__`
+    - With tenant: verifies membership, then returns tenant prefix, e.g. `t_<tenant>__`
+    """
+    app_state = get_app_state(request)
```

Comment:
What I would do here is always return the user prefix and return the tenant prefix if requested